### ISSUE TITLE
Training error in VIDSTG.

### ISSUE THREAD

**Issue by liujiaheng** [antoyang/TubeDETR#8]
![image](https://user-images.githubusercontent.com/27964642/174237016-05e3aad2-08c6-4614-a6c2-5c374a67877a.png)
![image](https://user-images.githubusercontent.com/27964642/174237145-6465ef71-25c8-4b5c-abfd-4377fe68c540.png)
Have you met this error?

**Comment by antoyang** [comment]
I haven't. You may verify that you use the suggested FFMPEG version, that you set up the path to the videos correctly in the config files and that the videos are properly downloaded.

**Comment by johnbager** [comment]
i have the same error as this. I think that could you list clearly the data catalog clearly?

**Comment by johnbager** [comment]
of course, i have run with the same ffmpeg and the pre code data successfuly.

**Comment by antoyang** [comment]
The folder "vidstg_vid_path" (in the config files) should contain a folder "video" in which there are the different folders containing VidSTG videos ("0000" etc).

**Comment by antoyang** [comment]
As I haven't met this error, I would suggest looking at which videos are problematic and what's the difference between len(images_list) and len(frame_ids) for these videos.

**Comment by liujiaheng** [comment]
Could you provide the download link of the ffmpeg (4.2.2 amd64 static)?
Besides, I also suppose the generated annotation json files after processing may be not accurate. Could you provide the generated json file?


Thanks.

**Comment by PeiqinZhuang** [comment]
> As I haven't met this error, I would suggest looking at which videos are problematic and what's the difference between len(images_list) and len(frame_ids) for these videos.

Hi, I also encountered some problems, which are the same as that of liujiang. Besides, the error also occurred in this place (https://github.com/antoyang/TubeDETR/blob/2f15abf78dadd254c3031823da595030abef28c8/engine.py#L107).  Since the authors added assertion here, I am not sure if you have filtered some failure videos. If possible, would you mind sharing your clean data list?

**Comment by TreezzZ** [comment]
Same problem.

> > As I haven't met this error, I would suggest looking at which videos are problematic and what's the difference between len(images_list) and len(frame_ids) for these videos.
> 
> Hi, I also encountered some problems, which are the same as those of liujiang. Besides, the error also occurred in this place .
> 
> https://github.com/antoyang/TubeDETR/blob/2f15abf78dadd254c3031823da595030abef28c8/engine.py#L107
> 
> Since the authors added assertion here, I am not sure if you have filtered some failure videos. If possible, would you mind sharing your clean data list?

Same problem. Have u solved it?

Thx.

**Comment by johnbager** [comment]
收到

**Comment by Awj2021** [comment]
> ![image](https://user-images.githubusercontent.com/27964642/174237016-05e3aad2-08c6-4614-a6c2-5c374a67877a.png) ![image](https://user-images.githubusercontent.com/27964642/174237145-6465ef71-25c8-4b5c-abfd-4377fe68c540.png) Have you met this error?

Hi, I met the same error, did you fix this problem? Thanks.

**Comment by salmank255** [comment]
I printed the length of frame_ids and image_list and got this:

AssertionError: len(images_list) 131
len(frame_ids) 130
for /video/1101/6797913724.mp4

This means one extra frame has been created by ffmpeg.

I fixed it by adding the below (conditions removing the extra frame) after this line https://github.com/antoyang/TubeDETR/blob/2f15abf78dadd254c3031823da595030abef28c8/datasets/vidstg.py#L115

"
        if len(images_list) > len(frame_ids):
            images_list = images_list[:len(frame_ids)]

        if len(frame_ids) > len(images_list):
            frame_ids = frame_ids[:len(images_list)]. "

**Comment by antoyang** [comment]
@liujiaheng @PeiqinZhuang I do not remember if this is the link I used for downloading it at the time but https://www.johnvansickle.com/ffmpeg/old-releases/ seems to provide the FFMPEG version I used for this project. I did not filter any video after the preprocessing script, the assertion was mainly for some initial checks. @salmank255 Did you use the FFMPEG version mentioned in the readme? In all problematic cases was it always one more frame created by FFMPEG?

**Comment by salmank255** [comment]
No, I used ffmpeg-python and the error of frames mismatch seems to be appeared only once after 200 iterations. Now, the next problem is the same facing by @PeiqinZhuang and @TreezzZ after 2700 iterations we are getting the same assertion error where the targets len is equal to 0.

**Comment by antoyang** [comment]
For the targets len 0, I am wondering if the videos are not corrupted. Can you ffprobe them to check? Otherwise a turnaround is to use FFMPEG to extract all frames of videos (5fps) once for all and to load each of them with PIL in the dataloader. It would make it easy to check if there is an issue with any video. I did this in an early version of the project and it was giving similar results.

**Comment by PeiqinZhuang** [comment]
> No, I used ffmpeg-python and the error of frames mismatch seems to have appeared only once after 200 iterations. Now, the next problem is the same faced by @PeiqinZhuang and @TreezzZ after 2700 iterations we are getting the same assertion error where the targets len is equal to 0.

Hi, you can just filter out those fault videos in advance. Unfortunately, no matter what version of ffmpeg I use, there are always some fault videos. You can use those codes in datasets, i.e. how to decode each video, to check every video before you run the code.

**Comment by salmank255** [comment]
Hi @PeiqinZhuang 

The faulty videos are 7003314632 and 9731127827. I removed both of them from the train annotation file and run the code now it's working fine.

**Comment by salmank255** [comment]
You also need to include below two conditions otherwise there will be an assertion error for frames lens

    if len(images_list) > len(frame_ids):
        images_list = images_list[:len(frame_ids)]
    if len(frame_ids) > len(images_list):
        frame_ids = frame_ids[:len(images_list)]
